### PR TITLE
[Reflection] Try to read only the reflection sections.

### PR DESCRIPTION
**Pull request by dcci**
(instead of the whole object).

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

File: include/swift/Reflection/ReflectionContext.h [hunk 1]
```diff
@@ -140,9 +140,49 @@ class ReflectionContext
       return false;
 
     // Read everything including the __TEXT segment.
-    Buf = this->getReader().readBytes(ImageStart, Command->vmsize);
+    // Buf = this->getReader().readBytes(ImageStart, Command->vmsize);
```
Comment (mikeash) [inline include/swift/Reflection/ReflectionContext.h]: Is this meant to be deleted?

**Review by mikeash — APPROVED** [review]

**Comment by dcci** [comment]
@mikeash unfortunately, this still crashes some testcases.

**Comment by dcci** [comment]
@swift-ci please test

**Pull request merged**